Sources/ServiceDiscovery/ServiceDiscovery+TypeErased.swift

```diff
@@ -176,6 +186,37 @@ public class AnyServiceDiscovery: ServiceDiscovery {
     }
 }
 
+#if compiler(>=5.5) && canImport(_Concurrency)
+public extension AnyServiceDiscovery {
+    /// See `lookup`.
+    ///
+    /// - Warning: If `Service` or `Instance` type does not match the underlying `ServiceDiscovery`'s associated types, it would result in a failure.
+    @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+    func lookupAndUnwrap<Service, Instance>(_ service: Service, deadline: DispatchTime? = nil) async throws -> [Instance] where Service: Hashable, Instance: Hashable {
```

Comment:
remind me the use case for the public "unwrap" functions?

Reply:
It's just for convenience.

I guess one can question whether the two impls in this file are used at all.